modules/installation-user-infra-machines-advanced-retain-disk.adoc

```diff
@@ -0,0 +1,131 @@
+// * installing/installing_bare_metal/upi/installing-bare-metal-network-customizations.adoc
+// * installing/installing_with_agent_based_installer/installing-with-agent-based-installer.adoc
+
+ifeval::["{context}" == "installing-restricted-networks-bare-metal"]
+:restricted:
+endif::[]
+
+:_mod-docs-content-type: REFERENCE
+[id="installation-user-infra-machines-advanced-retain-disk_{context}"]
+= Retaining existing partitions
+
+[role="_abstract"]
+For an ISO installation, you can add options to the `coreos-installer` command that causes the installation program to maintain one or more existing partitions. For a PXE installation, you can add `coreos.inst.*` options to the `APPEND` parameter to preserve partitions.
+
+Saved partitions might be data partitions from an existing {product-title} system. You can identify the disk partitions you want to keep either by partition label or by number.
+
+[NOTE]
+====
+If you save existing partitions, and those partitions do not leave enough space for {op-system}, the installation will fail without damaging the saved partitions.
+====
+
+The following examples preserves any existing partition during an ISP installation in which the partition label begins with `data` (`data*`):
+
+ifndef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partlabel 'data*' \
+/dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+ifdef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partlabel 'data*' \
+--offline \
+/dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+The following example illustrates running the `coreos-installer` in a way that preserves
+the sixth (6) partition on the disk:
+
+ifndef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partindex 6 /dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+ifdef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partindex 6 \
+--offline \
+/dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+This example preserves partitions 5 and higher:
+
+ifndef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partindex 5- /dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+ifdef::restricted[]
+[source,terminal]
+----
+# coreos-installer install --ignition-url http://10.0.2.2:8080/user.ign \
+--save-partindex 5- \
+--offline \
+/dev/disk/by-id/scsi-<serial_number>
+----
+endif::[]
+
+In the earlier examples where partition saving is used, `coreos-installer` recreates the partition immediately.
+
+The following examples preserves existign partitions during a PXE installation. Note that the `APPEND` option preserves any partition in which the partition label begins with 'data' ('data*'):
```

Comment:
`Note that the `APPEND` option preserves any partition in which the partition label begins with 'data' ('data*'):`
Needs to be an admonition.